test/ng/compileSpec.js

```diff
@@ -12111,6 +12111,620 @@ describe('$compile', function() {
+
+            // As a RESOURCE URL
+            element = $compile('<' + tag + ' ng-prop-src="testUrl"></' + tag + '>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('untrusted:foo()');
+            $rootScope.$digest();
+            expect(element.prop('src')).toEqual('untrusted:foo()');
+          }));
+        });
+      }
+    });
+
+    // Support: IE 9 only
+    // IE 9 rejects the `source` / `track` tags with
+    // "Unable to get value of the property 'childNodes': object is null or undefined"
+    if (msie !== 9) {
+      ['source', 'track'].forEach(function(tag) {
+        describe(tag + '[src]', function() {
+          it('should NOT require trusted values for whitelisted URIs', inject(function($rootScope, $compile) {
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = 'http://example.com/image.mp4'; // `http` is whitelisted
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('http://example.com/image.mp4');
+          }));
+
+          it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+            // As a MEDIA_URL URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsMediaUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+
+            // As a RESOURCE URL
+            element = $compile('<video><' + tag + ' ng-prop-src="testUrl"></' + tag + '></video>')($rootScope);
+            $rootScope.testUrl = $sce.trustAsResourceUrl('javascript:foo()');
+            $rootScope.$digest();
+            expect(element.find(tag).prop('src')).toEqual('javascript:foo()');
+          }));
+        });
+      });
+    }
+
+    describe('img[src] sanitization', function() {
+
+      it('should accept trusted values', inject(function($rootScope, $compile, $sce) {
+        element = $compile('<img ng-prop-src="testUrl"></img>')($rootScope);
+        // Some browsers complain if you try to write `javascript:` into an `img[src]`
+        // So for the test use something different
+        $rootScope.testUrl = $sce.trustAsMediaUrl('someuntrustedthing:foo();');
+        $rootScope.$digest();
+        expect(element.prop('src')).toEqual('someuntrustedthing:foo();');
+      }));
+
+      it('should sanitize concatenated values even if they are trusted', inject(function($rootScope, $compile, $sce) {
```

Comment:
Is this test and the next correct for expressions like this? I'm not sure if expressions vs interpolation are or should be handled different...?

Reply:
Delete?

Reply:
I think so. Was the consensus yesterday that ngProp can / should only consider the result of all expressions?